proposals/004-health-checks.md

```diff
@@ -0,0 +1,87 @@
+Syndesis - Camel Health Check
+=============================================
+
+* Issus:
+  * https://github.com/syndesisio/syndesis-project/issues/42
+  * https://issues.apache.org/jira/browse/CAMEL-11443
+
+## Background
+
+There is two kind of health checks required for Syndesis:
+
+* Technical health checks which ensure that the Integration itself is running, but not whether the components are without faults. This healthcheck is used as liveness and readiness checks for OpenShift
+* Status of an integration with respect to the backends. This status should be visualised in the Syndesis UI to give direct feedback to the user. It must not be the case that a faulty backend restarts the backend.
+
+### Technical health checks
+
+These are native OpenShift readiness and liveness probes.
+
+It is important that the readiness and liveness probes are only being used casually to ensure the integration can startup and become ready, and that the JVM stays alive. We can leverage the existing Camel health check from camel-spring-boot (`/health` HTTP REST endpoint), that ties into OpenShift readiness/liveness probes and ensures the JVM, Spring-Boot, and Apache Camel can startup and be ready. The livensss probe then calls the same `/health` HTTP REST endpoint which just checks that Spring Boot, and Camel is running.
+
+We do not want the OpenShift liveness probe to react on errors from the integration because when an integration is in an error state because of one of its connectors fails, this should be reported by other means than Kubernetes status changes. It should be visible on the Syndesis UI instead (i.e. integration health checks).
+
```

Comment:
In addition: we don't want Kuberenetes to constantly restart the pods when a backend fails because that doesn't help anyway.